### ISSUE TITLE
src is not set when using file

### ISSUE THREAD

**Issue by amacpherson-upsiide** [cookpete/react-player#1508]
#### Current Behavior

When setting URL = a mp4 the video SRC is never set, thus the player never loads. Occasionally when hot reloading kicks in the player will work.

#### Expected Behavior

setting `url="https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4"` should work

also tried, `url={[{src: 'https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4'}]}
`
#### Steps to Reproduce

- create react app
- import import ReactPlayer from 'react-player/file'; OR import ReactPlayer from 'react-player';
- add player
`<ReactPlayer
    url="https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4"
    controls
/>
`
![Screen Shot 2022-09-20 at 3 16 27 PM](https://user-images.githubusercontent.com/104508915/191345232-7a739b07-1b6e-47e4-97c2-1ba8ed9580d5.png)

**Comment by amacpherson-upsiide** [comment]
I can also see an error returned by the onError call back with the following...
`MEDIA_ELEMENT_ERROR: Empty src attribute`

**Comment by amacpherson-upsiide** [comment]
Downgraded CRA to react 17 and things seem to be working so chalking this up to an issue with React 18 specifically.

**Comment by cookpete** [comment]
Could this be a weird side effect of https://github.com/cookpete/react-player/pull/1360? Not sure what’s going on here.

**Comment by kayden-kim** [comment]
I have same issue too. I think that 'src' should be set again when component is re-mounted.

https://reactjs.org/docs/strict-mode.html#ensuring-reusable-state
> With Strict Mode starting in React 18, whenever a component mounts in development, React will simulate immediately unmounting and remounting the component

**Comment by usamamashkoor** [comment]
I am facing the same issue with React.js 18 and Next.js, did any one find a solution for this?

**Comment by rklf** [comment]
I was facing also this issue (React 18.2.0 - react-player 2.11.0)

Comment (or removing) ``<React.StrictMode>`` tag is fixing that in development environment (NODE_ENV), it is working in production environment (when app is built) though since ``<React.StrictMode>`` is ignored in production

**Comment by nickchauhan** [comment]
Downgrading react-player to v2.10.1 or removing `<React.StrictMode>` fixes the issue. I believe the problem is PR #1360.

**Comment by kevin-guertin** [comment]
I have the same issue in React 18. Downgrading to v2.10.1 does NOT fix the issue. Everything is fine after removing `<React.StrictMode>` as mentioned previously in both v2.10.1 and v2.11.0. However, that is not an acceptable fix.

In my function component, I do not render the video player at all if the URL string happens to be empty/null/undefined.

Here is part of the error that is generated from the `onError` callback. This same error happens in v2.10.1 and in v2.11.0 in React 18. You can see that my local video file path appears in the `currentSrc` property so it looks like the component is receiving the proper URL.

![image](https://user-images.githubusercontent.com/50712250/194571151-18db03fb-ce55-4f65-83d9-6746d816143b.png)

**Comment by furkan-cat** [comment]
> Downgrading react-player to v2.10.1 or removing `<React.StrictMode>` fixes the issue. I believe the problem is PR #1360.

Yep when i remove the strict mode all works fine. Is there any solution for this issue?

**Comment by Hydrock** [comment]
> Downgrading react-player to v2.10.1 or removing `<React.StrictMode>` fixes the issue. I believe the problem is PR #1360.

I can't downgrade react because i use NextJS last version, but downgrading react-player to v2.10.1 is helped me. Thank you gues a lot.

**Comment by KillerRP** [comment]
So when will it get fixed in react 18 and without downgrade any library or removing react strict mode

**Comment by cookpete** [comment]
Hopefully fixed by https://github.com/cookpete/react-player/pull/1538

**Comment by phcorp** [comment]
was not fixed by #1538 for me (I'm using `v2.11.2`)

the problem appears for urls provided as an array of objects with type and src properties
the problem does not appear when the URL is provided as a string

**Comment by jamesryan-dev** [comment]
I am also experiencing this issue:

<img width="420" alt="Screenshot 2023-03-20 at 10 10 48" src="https://user-images.githubusercontent.com/13485129/226309531-9a3c9874-1bde-421c-8baa-e7877e057d9d.png">

```
  return (
    <ReactPlayer
      //   url="https://www.youtube.com/watch?v=y9j-BL5ocW8"
      url={[
        { src: "/retail-video.webm", type: "video/webm" },
        { src: "/retail-video2.mp4", type: "video/mp4" },
      ]}
      loop={true}
      playing={true}
      controls={false}
      muted={true}
      width="100%"
      height="100%"
      config={{
        youtube: {
          playerVars: {
            start: "10",
          },
        },
      }}
    />
  );
```

**Comment by davshoward** [comment]
Also experiencing this. Single string works fine, array of either strings or objects fails.

**Comment by phcorp** [comment]
https://github.com/cookpete/react-player/pull/1612 supersedes this

**Comment by aayodejii** [comment]
The issue is yet to be resolved, I'm using 2.12.0
![image](https://github.com/cookpete/react-player/assets/65215959/43308608-666f-4f2f-b1e9-ec1465043e0d)

**Comment by otwm** [comment]
<img width="885" alt="스크린샷 2023-06-30 오후 8 49 24" src="https://github.com/cookpete/react-player/assets/136394/d38d2744-0169-44a9-a959-56b7762239de">
me too, The issue is yet to be resolved, I'm using 2.12.0

**Comment by mohamedanushkar** [comment]
Me too, cant play multiple urls for safari and chrome. anyone know s a fix?
